src/go/k8s/cli/configurator/main.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+// main package. This line should be remove after merging the following
+// https://github.com/vectorizedio/redpanda/pull/753
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/spf13/afero"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	hostName		string
+	svcFQDN			string
+	configSourceDir		string
+	configDestination	string
+	redpandaRPCPort		int
+)
+
+var errorMissingEnvironmentVariable = errors.New("missing environment variable")
+
+func main() {
+	if err := checkEnvVars(); err != nil {
+		log.Fatalf("%s", fmt.Errorf("unable to get the environment variables: %w", err))
+	}
+
+	fs := afero.NewOsFs()
+	v := config.InitViper(fs)
+	v.AddConfigPath(configSourceDir)
+
+	if err := v.ReadInConfig(); err != nil {
+		log.Fatalf("%s", fmt.Errorf("unable to read the redpanda configuration file: %w", err))
+	}
+
+	cfg := &config.Config{}
+	decoderConfig := mapstructure.DecoderConfig{
+		Result:	cfg,
+		// Sometimes viper will save int values as strings (i.e.
+		// through BindPFlag) so we have to allow mapstructure
+		// to cast them.
+		WeaklyTypedInput:	true,
```

Comment:
Nice

Reply:
All credits to @0x5d :) https://github.com/vectorizedio/redpanda/blame/dev/src/go/rpk/pkg/config/manager.go#L403